problems/problems.go

```diff
@@ -0,0 +1,74 @@
+package errors
+
+import (
+	"fmt"
+)
+
+//Problem implements the RFC7807 "problem"/error standard
+//Additional error field can be defined in custom struct implementing the Problem struct
+type Problem struct {
+	Title    string `json:"title"`
+	Detail   string `json:"detail,omitempty"`
+	Type     string `json:"type,omitempty"`
+	Instance string `json:"instance,omitempty"`
+	Status   int    `json:"status,omitempty"`
+}
+
+//Error implements the error interface
+func (p *Problem) Error() string {
```

Comment:
Also another thought.
Accessing the pointer means the object itself is not imutable.
I would rather make the methods value receivers in a builder pattern. This way we ensure thread safety when the error might be edited after processing or passed through different threads.